src/test/java/FileUtilityTest.java

```diff
@@ -243,4 +280,553 @@ public void testSetPermissions() throws IOException {
+            Assertions.assertTrue(eventsForRegisteredPath.contains(StandardWatchEventKinds.ENTRY_CREATE));
+        }
+
+        // Deregister sub path
+        Path pathToDeRegister = testBedPath.resolve(pathToDeRegisterString);
+        FileUtility.deRegisterWatchServiceForDirectory(pathToDeRegister);
+        // Perform actions in sub path
+        Files.createFile(pathToDeRegister.resolve("random.txt"));
+        Thread.sleep(1_000);
+        Queue<WatchEvent.Kind<?>> eventsForDeRegisteredPath = actualPathToEventKindMap.get(
+                pathToDeRegister.toAbsolutePath()
+                        .toString()
+        );
+        // Assert actions have not been logged
+        Assertions.assertNull(eventsForDeRegisteredPath);
+        clearTestBed();
+    }
+
+    /**
+     * Test register events with only one type of event.
+     * @param eventToRegisterString Event to register
+     * @throws IOException Thrown if registering a WatchService or walking through a path  or modifying it fails
+     * @throws InterruptedException Thrown if the current thread is interrupted
+     */
+    @ParameterizedTest
+    @CsvSource(
+        {
+            "ENTRY_CREATE", // Create event
+            "ENTRY_MODIFY", // Modify event
+            "ENTRY_DELETE", // Delete event
+        }
+    )
+    public void testSelectiveRegisteringWatchService( String eventToRegisterString) throws IOException, InterruptedException {
+        actualPathToEventKindMap.clear();
+        createTestBed();
+
+        WatchEvent.Kind<?>  eventToRegister = switch(eventToRegisterString){
+            case("ENTRY_CREATE") ->  StandardWatchEventKinds.ENTRY_CREATE;
+            case("ENTRY_MODIFY") -> StandardWatchEventKinds.ENTRY_MODIFY;
+            case("ENTRY_DELETE") -> StandardWatchEventKinds.ENTRY_DELETE;
+            default -> throw new IllegalArgumentException("Invalid eventToRegisterString - " + eventToRegisterString);
+        };
+
+        FileUtility.registerWatchServiceIfPathIsADirectoryAndIsNotAlreadyRegistered(
+                testBedPath,
+                3,
+                (path, watchEvent) -> {
+                    logger.info("CONSUMER: Event kind:" + watchEvent.kind() + ". File affected: " + path + ".");
+                    String absolutePathString = path.toAbsolutePath().toString();
+                    actualPathToEventKindMap.putIfAbsent(absolutePathString, new LinkedList<>());
+                    actualPathToEventKindMap.get(absolutePathString)
+                            .add(watchEvent.kind());
+                },
+                eventToRegister
+        );
+        // Perform the actions to trigger the events
+        Thread.sleep(2_000);
+        Path fileToCreatePath = testBedPath.resolve("selectiveRegistering.txt");
+        Files.createFile(fileToCreatePath);
+        Files.setLastModifiedTime(fileToCreatePath, FileTime.from(Instant.now()));
+        Files.delete(fileToCreatePath);
+        Thread.sleep(2_000);
```

Comment:
Can we put this as an entry into test.properties?

Reply:
ddone